saleor/tests/e2e/shop/utils/shop_update_settings.py

```diff
@@ -18,8 +18,16 @@
 """
 
 
-def update_shop_settings(staff_api_client, input):
-    variables = {"input": input}
+def update_shop_settings(
+    staff_api_client,
+    input_data={
```

Comment:
It would be safer, to pass here all settings flag direct (without `input_data`). 
(see my previous comment about mutable list)
```python
def update_shop_settings(
     staff_api_client,
     enable_account_confirmation_by_email=True,
     ...):
     variables = {"input": {"enableAccountConfirmationByEmail": enable_account_confirmation_by_email, ...}
```

Reply:
I tried to implement your suggestion but I think the code would be less readable:

`shop_update_settings.py`
````
SHOP_SETTING_UPDATE_MUTATION = """
mutation ShopSettingsUpdate($input: ShopSettingsInput!) {
  shopSettingsUpdate(input: $input) {
    errors {
      field
      message
      code
    }
    shop {
      enableAccountConfirmationByEmail
      fulfillmentAutoApprove
      fulfillmentAllowUnpaid
    }
  }
}
"""

def update_shop_settings(
    staff_api_client,
    enable_account_confirmation_by_email=None,
    allow_login_without_confirmation=None,
    fulfillment_auto_approve=None,
    fulfillment_allow_unpaid=None,
):
    if enable_account_confirmation_by_email is None:
        enable_account_confirmation_by_email = True
    if allow_login_without_confirmation is None:
        allow_login_without_confirmation = False
    if fulfillment_auto_approve is None:
        fulfillment_auto_approve = False
    if fulfillment_allow_unpaid is None:
        fulfillment_allow_unpaid = False
        
    variables = {
        "input":{
            "enableAccountConfirmationByEmail": enable_account_confirmation_by_email,
            "allowLoginWithoutConfirmation": allow_login_without_confirmation,
            "fulfillmentAutoApprove": fulfillment_auto_approve,
            "fulfillmentAllowUnpaid": fulfillment_allow_unpaid,
            }
        }
```

`preparing_shop.py`
```
def prepare_shop(
    e2e_staff_api_client,
    channels=None,
    shop_settings=None,
    tax_settings=None,
):    
if shop_settings is not None:
        update_shop_settings(
            e2e_staff_api_client,
            enable_account_confirmation_by_email=shop_settings.get(
                "enable_account_confirmation_by_email", None
            ),
            allow_login_without_confirmation=shop_settings.get(
                "allow_login_without_confirmation", None
            ),
            fulfillment_auto_approve=shop_settings.get(
                "fulfillment_auto_approve", None
            ),
            fulfillment_allow_unpaid=shop_settings.get(
                "fulfillment_allow_unpaid", None
            ),
        )
...
```
Unless I am not aware of a better solution - please provide one if you have it so I could understand the reason behind the requested change @korycins

Reply:
Unless I am not aware of a better solution - please provide one if you have it so I could understand the reason behind the requested change @korycins